From 4d63520dde51d94fd9563cf292a71edc49cfa04e Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 7 May 2011 04:03:58 +0100
Subject: [PATCH] ldm: Disable broken support for VBLK fragments

ldm_frag_add() has various bugs which can be exploited with a
carefully constructed partition table to cause denial-of-service or
privilege escalation.  Further, it doesn't appear to work for any
valid fragmented VBLKs either.  Disable it until a fix is available.
---
 fs/partitions/ldm.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/fs/partitions/ldm.c b/fs/partitions/ldm.c
index b10e354..d963c86 100644
--- a/fs/partitions/ldm.c
+++ b/fs/partitions/ldm.c
@@ -1293,6 +1293,11 @@ static bool ldm_ldmdb_add (u8 *data, int len, struct ldmdb *ldb)
  */
 static bool ldm_frag_add (const u8 *data, int size, struct list_head *frags)
 {
+#if 1
+	/* This function doesn't work */
+	ldm_error("Unsupported fragmented VBLK.");
+	return false;
+#else
 	struct frag *f;
 	struct list_head *item;
 	int rec, num, group;
@@ -1341,6 +1346,7 @@ found:
 	memcpy (f->data+rec*(size-VBLK_SIZE_HEAD)+VBLK_SIZE_HEAD, data, size);
 
 	return true;
+#endif
 }
 
 /**
-- 
1.7.4.4

